src/repository.py

```diff
@@ -48,6 +96,92 @@ def __init__(self, repository: Repo, github_repository: Repository) -> None:
         self._github_repo = github_repository
         self._configure_git_user()
 
+    @cached_property
+    def base_path(self) -> Path:
+        return Path(self._git_repo.working_dir)
+
+    @property
+    def metadata(self) -> Metadata:
+        return get_metadata(self.base_path)
+
+    @property
+    def has_docs_directory(self) -> bool:
+        return has_docs_directory(self.base_path)
+
+    @property
+    def current_branch(self) -> str:
+        return self._git_repo.active_branch.name
+
+    @contextmanager
+    def with_branch(self, branch_name: str) -> 'Client':
+        current_branch = self.current_branch
+
+        yield self.switch(branch_name)
```

Comment:
Should this be wrapped in a `try`, `finally`?

Reply:
I don't think so. I mean, this `yield` is to easily define contexts (together with the decorator `contextmanager`), that they can work with 

```
with with_branch(...) as repo:
     # use repo
```

not to catch exception with try-except-finally blocks.

Reply:
The thought process was that we might want to make sure that the branch is switched back even if an exception occurs during yield so something like

```python
try:
    yield ...
finally:
    # switch branch back
```

So no exception handling, just ensuring that the switch back is executed even if an exception occurs. I haven't done that before, not sure if it works